lib/IHP/static/ihp-schemadesigner.js

```diff
@@ -1,3 +1,11 @@
+function toSingular(word) {
```

Comment:
When using a reserved keyword, the `is-invalid` class is added to the text field, but displaying the error message fails while using `toSingular(word)`:
```
Uncaught TypeError: word is undefined
```
In general, we use `singularize` from Text.Countable. Using this as the data attribute directly would allow to remove the JS singularize function.

Reply:
Avoiding JS is a good idea in general.

Reply:
Avoiding JS is everything we stand for. I just tested everything and it seems fine; Good job!